src/PrpShop/PRP/Render/QPlasmaCubicRender.cpp

```diff
@@ -0,0 +1,412 @@
+
+    fGLArrayBuf.destroy();
+    fGLIndexBuf.destroy();
+
+    delete fGLTexture;
+    delete fGLLabelTexture;
+
+    doneCurrent();
+
+    disconnect(context(), &QOpenGLContext::aboutToBeDestroyed, this, &QPlasmaCubicRender::cleanupGL);
+}
+
+void QPlasmaCubicRender::mouseMoveEvent(QMouseEvent *evt)
+{
+    if (evt->buttons() & Qt::LeftButton) {
+        QPoint currentPos = evt->pos();
+
+        // Get screen geometry and current global cursor position
+        QRect screenGeometry = QApplication::primaryScreen()->geometry();
+        QPoint center = mapFromGlobal(screenGeometry.center());
+        QPoint globalPos = mapToGlobal(currentPos);
+
+        // Check if cursor hit screen bounds
+        if (globalPos.x() <= screenGeometry.left() ||
+            globalPos.x() >= screenGeometry.right() ||
+            globalPos.y() <= screenGeometry.top() ||
+            globalPos.y() >= screenGeometry.bottom()) {
+
+            // Calculate the center point and warp cursor there
+            QCursor::setPos(mapToGlobal(center));
+            currentPos = center;
+
+            // Since we warped, reset fMouseFrom as well or else we'll get a jump
+            fMouseFrom = currentPos;
+        }
+
+        // Convert pixel movement to angles (and scale down the movement)
+        QPoint diff = currentPos - fMouseFrom;
+        float xAngle = diff.y() * -0.5f;  // Pitch
+        float yAngle = diff.x() *  0.5f;  // Yaw
+
+        // Create rotation quaternions for each axis
+        QQuaternion xRotation = QQuaternion::fromAxisAndAngle(1.0f, 0.0f, 0.0f, xAngle);
+        QQuaternion yRotation = QQuaternion::fromAxisAndAngle(0.0f, 1.0f, 0.0f, yAngle);
+        
+        // Apply the new rotation, mindful of ordering to prevent roll
+        fGLRotation = xRotation * fGLRotation * yRotation;
+
+        // Update the stored position
+        fMouseFrom = currentPos;
+
+        // Request a redraw
+        update();
+    }
+}
+
+void QPlasmaCubicRender::mousePressEvent(QMouseEvent *evt)
+{
+    if (evt->button() == Qt::LeftButton) {
+        fMouseFrom = evt->pos();
+        setCursor(Qt::BlankCursor);  // Hide the cursor when starting a drag
+    }
+}
+
+void QPlasmaCubicRender::mouseReleaseEvent(QMouseEvent *evt)
+{
+    if (evt->button() == Qt::LeftButton) {
+        setCursor(Qt::ArrowCursor);  // Restore the cursor when done dragging
+    }
+}
+
+void QPlasmaCubicRender::buildFaceLabelTexture()
+{
+    // Create a texture to hold 3x2 cube faces
+    int face_size = 512;
+    fFaceLabelTexture = QImage(QSize(face_size * 3, face_size * 2), QImage::Format_ARGB32);
+    fFaceLabelTexture.fill(Qt::transparent);
+
+    QPainter painter(&fFaceLabelTexture);
+    painter.setRenderHint(QPainter::Antialiasing);
+    painter.setPen(QPen(Qt::black));
+
+    // Set font size to a reasonable percentage of the texture resolution
+    QFont font = painter.font();
+    font.setPixelSize(face_size / 8);
+    painter.setFont(font);
+
+    QFontMetrics fm(font);
```

Comment:
Since we're rendering with GL using floating point coordinates, I wonder if it's worth using the more precise `QFontMetricsF` here instead...